crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -669,6 +669,23 @@ fn unknown_utd_reason() -> UnableToDecryptReason {
     UnableToDecryptReason::Unknown
 }
 
+/// A room key might be missing because the sender refused to share it or was
+/// not technically able to share it.
+/// In the protocol this is reflected by clients sending `m.room_key.withheld`
+/// to the participant that won't receive the key.
+/// We only want here to consider the subset of codes that clients should
+/// use to display specific UX.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
+pub enum WithheldReason {
+    /// When the sender refuses to send the key because his security settings in
+    /// that room are not met by your device. Could be that your device is
+    /// not cross-signed (insecure device) or that you are not verified by
+    /// the sender.
+    TrustRequirementMismatch,
+    /// Other reasons. Like fail to establish a secure channel to share the key,
+    /// or if for some reason our device is blocked.
+    Other,
+}
```

Comment:
Could we instead move the existing `WithheldCode` into matrix-sdk-common, rather than inventing a new type that does basically the same thing?

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/4305/commits/e3b6c5007a8480be414c58717cff6aadd2c042e3
and
https://github.com/matrix-org/matrix-rust-sdk/pull/4305/commits/12bb41daf819be85360610e529a4ec5c5385adc4